trace-mini-agent/src/mini_agent.rs

```diff
@@ -27,11 +29,27 @@ pub struct MiniAgent {
     pub trace_flusher: Arc<dyn trace_flusher::TraceFlusher + Send + Sync>,
     pub stats_processor: Arc<dyn stats_processor::StatsProcessor + Send + Sync>,
     pub stats_flusher: Arc<dyn stats_flusher::StatsFlusher + Send + Sync>,
+    pub env_verifier: Arc<dyn env_verifier::EnvVerifier + Send + Sync>,
 }
 
 impl MiniAgent {
     #[tokio::main]
     pub async fn start_mini_agent(&self) -> Result<(), Box<dyn std::error::Error>> {
+        // verify we are in a google cloud funtion environment. if not, shut down the mini agent.
```

Comment:
my understanding is that we're not blocking on this internal metadata api calls which could be an issue if we're receiving telemetry meanwhile, they won't be tagged properly. WDYT about making this a blocking call (with a short timeout value + default value if timeout exceeded or errored?